great_expectations/checkpoint/util.py

```diff
@@ -239,76 +272,247 @@ def get_runtime_batch_request(
-        "expectation_suite_ge_cloud_id": validation_dict.get(
+def substitute_template_config(source_config: dict, template_config: dict) -> dict:
+    if not (template_config and any(template_config.values())):
+        return source_config
+
+    dest_config: dict = copy.deepcopy(template_config)
+
+    # replace
+    if source_config.get("name") is not None:
+        dest_config["name"] = source_config["name"]
+    if source_config.get("module_name") is not None:
+        dest_config["module_name"] = source_config["module_name"]
+    if source_config.get("class_name") is not None:
+        dest_config["class_name"] = source_config["class_name"]
+    if source_config.get("run_name_template") is not None:
+        dest_config["run_name_template"] = source_config["run_name_template"]
+    if source_config.get("expectation_suite_name") is not None:
+        dest_config["expectation_suite_name"] = source_config["expectation_suite_name"]
+    if source_config.get("expectation_suite_ge_cloud_id") is not None:
+        dest_config["expectation_suite_ge_cloud_id"] = source_config[
             "expectation_suite_ge_cloud_id"
+        ]
+
+    # update
+    if source_config.get("batch_request") is not None:
+        batch_request = dest_config.get("batch_request") or {}
+        updated_batch_request = nested_update(
+            batch_request,
+            source_config["batch_request"],
         )
-        or substituted_runtime_config.expectation_suite_ge_cloud_id,
-        "action_list": CheckpointConfig.get_updated_action_list(
-            base_action_list=substituted_runtime_config.action_list,
-            other_action_list=validation_dict.get("action_list", {}),
-        ),
-        "evaluation_parameters": nested_update(
-            substituted_runtime_config.evaluation_parameters,
-            validation_dict.get("evaluation_parameters", {}),
-        ),
-        "runtime_configuration": nested_update(
-            substituted_runtime_config.runtime_configuration,
-            validation_dict.get("runtime_configuration", {}),
-        ),
-    }
-    if validation_dict.get("name") is not None:
-        substituted_validation_dict["name"] = validation_dict["name"]
-    validate_validation_dict(substituted_validation_dict)
-    return substituted_validation_dict
+        dest_config["batch_request"] = updated_batch_request
+    if source_config.get("action_list") is not None:
+        action_list = dest_config.get("action_list") or []
+        dest_config["action_list"] = get_updated_action_list(
+            base_action_list=action_list,
+            other_action_list=source_config["action_list"],
+        )
+    if source_config.get("evaluation_parameters") is not None:
+        evaluation_parameters = dest_config.get("evaluation_parameters") or {}
+        updated_evaluation_parameters = nested_update(
+            evaluation_parameters,
+            source_config["evaluation_parameters"],
+        )
```

Comment:
Similar pattern here?

Reply:
@cdkini Could you please put this idea into a separate JIRA ticket (along with the others above) for us to follow up on?  Thanks!